hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/test/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/KafkaBackendResolverTest.java

```diff
@@ -0,0 +1,66 @@
+package org.hypertrace.traceenricher.enrichment.enrichers.resolver.backend;
+
+import static org.mockito.Mockito.mock;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Map;
+import org.hypertrace.core.datamodel.AttributeValue;
+import org.hypertrace.core.datamodel.Attributes;
+import org.hypertrace.core.datamodel.Event;
+import org.hypertrace.core.datamodel.EventRef;
+import org.hypertrace.core.datamodel.EventRefType;
+import org.hypertrace.core.datamodel.MetricValue;
+import org.hypertrace.core.datamodel.Metrics;
+import org.hypertrace.core.datamodel.shared.StructuredTraceGraph;
+import org.hypertrace.entity.data.service.v1.Entity;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+
+/**
+ * Unit Test for {@link KafkaBackendResolver}
+ */
+public class KafkaBackendResolverTest {
+
+  private KafkaBackendResolver kafkaBackendResolver;
+  private StructuredTraceGraph structuredTraceGraph;
+
+  @BeforeEach
+  public void setup() {
+    kafkaBackendResolver = new KafkaBackendResolver();
+    structuredTraceGraph = mock(StructuredTraceGraph.class);
+  }
+
+  @Test
+  public void TestBackendEventResolution() {
+    String broker = "kafka-broker.hypertrace.com";
+    Entity entity = kafkaBackendResolver.resolveEntity(getKafkaBackendEvent(broker), structuredTraceGraph).get();
+    Assertions.assertEquals(broker, entity.getEntityName());
+
+  }
+
+  private Event getKafkaBackendEvent(String broker) {
+    Event event =  Event.newBuilder().setCustomerId("customer1")
+        .setEventId(ByteBuffer.wrap("bdf03dfabf5c70f9".getBytes()))
+        .setEntityIdList(Arrays.asList("4bfca8f7-4974-36a4-9385-dd76bf5c8824")).setEnrichedAttributes(
+            Attributes.newBuilder().setAttributeMap(
+                Map.of("SPAN_TYPE", AttributeValue.newBuilder().setValue("EXIT").build())).build())
+        .setAttributes(Attributes.newBuilder().setAttributeMap(Map
+            .of("messaging.system", AttributeValue.newBuilder().setValue("kafka").build(),
+                "messaging.url", AttributeValue.newBuilder().setValue(broker).build(),
```

Comment:
try `messaging.url` with proper url - `xyz://localhost:9092`, I think, that is what we will receive here as given example here - https://github.com/open-telemetry/opentelemetry-specification/blob/1afab39e5658f807315abf2f3256809293bfd421/specification/trace/semantic_conventions/messaging.md#messaging-attributes

Reply:
Made the change. Now the test is to check hostname with entity name.